File: sdk/ruby/test/wire/agent_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class AgentWireTest < WireMockTestCase
  def setup
    super

    @client = AgentMailer::Client.new(
      api_key: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_agent_bootstrap_with_wiremock
    test_id = "agent.bootstrap.0"

    @client.agent.bootstrap(request_options: {
      additional_headers: {
        "X-Test-Id" => "agent.bootstrap.0"
      }
    })

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/agent/bootstrap",
      query_params: nil,
      expected: 1
    )
  end

  def test_agent_sign_up_with_wiremock
    test_id = "agent.sign_up.0"

    @client.agent.sign_up(
      human_email: "human_email",
      username: "username",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "agent.sign_up.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/agent/sign-up",
      query_params: nil,
      expected: 1
    )
  end
end

File: sdk/rust/tests/agent_test.rs
use agentmailer::prelude::*;

mod wire_test_utils;

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_agent_bootstrap_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client
        .agent
        .bootstrap(
            &BootstrapRequest {
                ..Default::default()
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("POST", "/v1/agent/bootstrap", None, 1)
        .await
        .unwrap();
}

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_agent_sign_up_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = AgentMailer::new(config).expect("Failed to build client");

    let result = client
        .agent
        .sign_up(
            &SignUpAgentRequest {
                human_email: "human_email".to_string(),
                username: "username".to_string(),
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count("POST", "/v1/agent/sign-up", None, 1)
        .await
        .unwrap();
}
